lib/view_component/base.rb

```diff
@@ -281,8 +283,26 @@ def validate_collection_parameter!(validate_default: false)
         )
       end
 
+      # Ensure the component initializer does not define
+      # invalid parameters that could override the framework's
+      # methods.
+      def validate_initialization_parameters!
+        invalid_parameters = initialize_parameter_names & BLACKLISTED_PARAMETERS
+
+        return unless invalid_parameters.any?
+
+        raise ArgumentError.new(
+          "#{self} initializer must not have " \
+          "`#{invalid_parameters.join(", ")}` as parameters."
```

Comment:
Thoughts on elaborating here? Maybe something similar to `initializer cannot contain #{invalid_parameters.join(", ")} since they will override public ViewComponent  methods`?